lib/auth.coffee

```diff
@@ -215,6 +215,41 @@ getAuth = (deps, opts) ->
 		.catchReturn(false)
 		.asCallback(callback)
 
+	###*
+	# @summary Generate a new API key for user
+	# @name generateApiKey
+	# @public
+	# @function
+	# @memberof resin.auth
+	#
+	# @description This will only work if you user is logged in.
+	#
+	# @fulfil {String} - raw API key
+	# @returns {Promise}
+	#
+	# @example
+	# resin.auth.generateApiKey().then(function(apiKey) {
+	# 	console.log(apiKey);
+	# });
+	#
+	# @example
+	# resin.auth.generateApiKey(function(error, apiKey) {
+	# 	if (error) throw error;
+	# 	console.log(apiKey);
+	# });
+	###
+	exports.generateApiKey = (callback) ->
+		request.send
+			method: 'POST'
+			url: '/api-key/user/full'
+			baseUrl: apiUrl
+			body:
+				name: 'apiKey'
```

Comment:
Here you're naming the api key, and this name is what will appear in the UI when the user browses their registered api keys.

We shouldn't use a default `apiKey` name for that - instead this method should take a required name parameter, and we should pass that to the API.